comid-triples.cddl

```diff
@@ -0,0 +1,172 @@
+concise-mid-tag = {
+  ; TODO(tho) language
+  ; TODO(tho) tag-id
+  ; TODO(tho) module-name
+  ; TODO(tho) entity
+  ; TODO(tho) linked-tags
+  comid.triples => triples-map
+  ; TODO(tho) extension
+}
+
+triples-map = non-empty<{
+  ? comid.reference-triples => one-or-more<reference-triple-record>
+  ? comid.endorsed-triples => one-or-more<endorsed-triple-record>
+  ? comid.attest-key-triples => one-or-more<attest-key-triple-record>
+  ? comid.identity-triples => one-or-more<identity-triple-record>
+  * $$triples-map-extension
+}>
+
+; REFERENCE-VALUE triple
+; "target environment $TE" "has reference measurements" "$RV"
+; NOTE(tho): can an attesting evironement have reference values?
+reference-triple-record = [
+  environment-map ; target environment
+  one-or-more<measurement-map> ; reference measurements
+]
+
+; ENDORSED-VALUE triple
+; "environment $E" "has endorsed measurements" "$EV"
+endorsed-triple-record = [
+  environment-map ; (target or attesting) environment
+  one-or-more<measurement-map> ; endorsed measurements
+]
+
+; ATTESTATION-VERIFICATION-KEY triple
+; "attesting environment $AE" "signs Evidence that can be verified with key" "$K"
+attest-key-triple-record = [
+  environment-map ; attesting environment
+  one-or-more<verification-key> ; attestation verification key(s)
+]
+
+; DEVICE-IDENTITY triple
+; "device $D" "is identified by key" "$K"
+identity-triple-record = [
+  environment-map ; device identifier (instance or class)
+  one-or-more<verification-key> ; DevID, or semantically equivalent
+]
+
+verification-key = {
+  ; Verification key (typically a public key)
+  comid.key => COSE_Key
+  ; Optional X.509 certificate chain corresponding to the public key
+  ; in comid.key, encoded as an array of one or more base64-encoded
+  ; DER PKIX certificates.  The certificate containing the public key
+  ; in comid.key MUST be the first certificate.  This MAY be followed
+  ; by additional ; certificates, with each subsequent certificate
+  ; being the one used to certify the previous one.
+  ? comid.keychain => [ + tstr ]
```

Comment:
shouldn't 'tstr' be 'certificate-type' or 'pkix-cert-type'?

Reply:
clarifying question: do you want an alias type for the PEM string?

Reply:
Is the question whether or not PEM encoding should be allowed in addition to base64?

Reply:
PEM *is* a base64 encoded DER certificate

Reply:
so the elements in the `comid.keychain` array are PEM encoded DER X.509 certs

Reply:
so maybe it is:
comid.keychain => [ + pkix-pem-cert-type ]
pkix-pem-cert-type = tstr

Reply:
> Alias identifiers could be addressed with a new triple where subject is the alias of object. The design trade-off is more triples vs. more flexible but fewer triples.
> […](#)

I am in favour of an explicit group-id rather than leaving a wildcard extension point.

Reply:
I think that works for group-id. EPID uses a 128-bit gid.